core/GhcMod/Cradle.hs

```diff
@@ -127,7 +132,16 @@ cabalCradle cabalProg wdir = do
           , cradleTempDir    = error "tmpDir"
           , cradleCabalFile  = Just cabalFile
           , cradleDistDir    = "dist"
+          , cradleGhcVersion = ghcVer
           }
+  where
+    tryProjectVer cabalDir = do
+      hasCabalProject <- liftIO $ doesFileExist $ cabalDir </> "cabal.project"
```

Comment:
`cabal.project` files are not required to reside in the directory where cabal is. They can be in any parent directory, see https://github.com/haskell/cabal